beacon-chain/slasher/service.go

```diff
@@ -121,6 +121,25 @@ func (s *Service) run() {
 	indexedAttsChan := make(chan *types.WrappedIndexedAtt, 1)
 	beaconBlockHeadersChan := make(chan *ethpb.SignedBeaconBlockHeader, 1)
 
+	// This section can be totally removed once Electra is on mainnet.
+	headSlot := s.serviceCfg.HeadStateFetcher.HeadSlot()
+	headEpoch := slots.ToEpoch(headSlot)
+
+	maxPruningEpoch := primitives.Epoch(0)
+	if headEpoch >= s.params.historyLength {
+		maxPruningEpoch = headEpoch - s.params.historyLength
+	}
+
+	// For database performance reasons, database read/write operations
+	// are chunked into batches of maximum `batchSize` elements.
+	const migrationBatchSize = 100_000
+
+	err = s.serviceCfg.Database.Migrate(s.ctx, headEpoch, maxPruningEpoch, migrationBatchSize)
```

Comment:
Does this perform the migrate every time the client starts?

Reply:
Yes.

Reply:
Yes, but if there is nothing to migrate (no epoch/slot higher than the current epoch/slot), then this functions exits immediately. Finding the highest epoch/slot in the DB is `O(1)`.

Reply:
OK, thanks